### DISCUSSION TITLE
Ads suddenly stopped

### DISCUSSION THREAD

**Discussion by ngdevr** [admob-plus/admob-plus#581] [Q&A]
I created one test app in the morning and one 12 hours later yesterday. The first one shows test ads but the second one does not. I created a few more afterwards to test and it seems like none of the new apps are showing ads but the one created in the morning yesterday continues to work. I do see some changes committed yesterday between my two test apps but not sure what actually broke my setup. Can someone confirms that this still works for them for a new app created today? Any idea on what to look for?

**Comment by ngdevr** [comment]
I do see the PLAY_SERVICES_VERSION updated from 20.6.0 to 22.1.0 between my two apps. Just want to confirm that this update has not broken anything and things are working like before. Do we have a dependency on a specific version of cordova?

**Comment by ratson** [comment]
Which version is installed? Did you try setting `PLAY_SERVICES_VERSION` back to `20.6.0`, and check if it works?

**Reply by ngdevr** [reply]
I have just created a new test app to test few scenarios. I have cordova-android 11.0.0  and targetSDkVersion for the app is 32. 

Cordova 12.0.0 is just released but not upgraded to it yet since its kinda new.** 

When I create a new app and import the latest version of admob-plus, ads don't work.
When I create a new app and import the latest version of admob-plus but change the PLAY_SERVICES_VERSION to 20.6.0, ads don't work.

I was not sure whether or not versioning is working in admob-plus, but I tried to import previous version of the plugin by using admob-plus-cordova@1.28.0, and this seems to work and load ads. I checked PLAY_SERVICES_VERSION in the plugin config which was 20.6.0.

I have tried to debug the app also to see where exactly in the plugin code, the app is failing. This is not throwing any error. All ad launch code is going through but no ad is loading and none of the callback functions 'load', 'loadfail', 'show', 'dismiss' etc get called. Not well-versed with java code to debug further. Let me know if you need any further information.

**Reply by ratson** [reply]
@ngdevr Please try the latest version `2.0.0-alpha.3`, which fixed a bug in showing banner ads.
Also, check the [upgrade guide](https://admob-plus.github.io/docs/cordova/upgrade/) to see if your code need to be changed.

**Reply by ngdevr** [reply]
Thanks, I will check this and get back with results. I do think my previous code will need to be changed.

Edit: I followed the upgrade guide to enable Kotlin. and testing now with very basic setup (just the admob-plus calls). Banner ads are working now and loading properly but interstitial still fails. interstitial .load() goes through but interstitial .show() immediately crashes the app. I did look into interstitial object after interstitial .load() was called and I see following message in the caller property

TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them
    at Function.invokeGetter (<anonymous>:3:28)
    at HTMLInputElement.showInterstitialFunc (https://localhost/js/index.js:126:17)

Could it be because of some missing configuration? or may be some deprecated calls being made in the library?
